traffic_ops/testing/api/v4/federation_resolvers_test.go

```diff
@@ -259,3 +265,126 @@ func DeleteTestFederationResolvers(t *testing.T) {
+	}
+
+	opts.QueryParameters.Set("limit", "1")
+	fedResolverWithLimit, _, err := TOSession.GetFederationResolvers(opts)
+	if err != nil {
+		t.Fatalf("cannot Get FederationResolver with Limit: %v - alerts: %+v", err, fedResolverWithLimit.Alerts)
+	}
+	if !reflect.DeepEqual(fedResolver[:1], fedResolverWithLimit.Response) {
+		t.Error("expected GET FederationResolver with limit = 1 to return first result")
+	}
+
+	opts.QueryParameters.Set("offset", "1")
+	fedResolverWithOffset, _, err := TOSession.GetFederationResolvers(opts)
+	if err != nil {
+		t.Fatalf("cannot Get FederationResolver with Limit and Offset: %v - alerts: %+v", err, fedResolverWithOffset.Alerts)
+	}
+	if !reflect.DeepEqual(fedResolver[1:2], fedResolverWithOffset.Response) {
+		t.Error("expected GET FederationResolver with limit = 1, offset = 1 to return second result")
+	}
+
+	opts.QueryParameters.Del("offset")
+	opts.QueryParameters.Set("page", "2")
+	fedResolverWithPage, _, err := TOSession.GetFederationResolvers(opts)
+	if err != nil {
+		t.Fatalf("cannot Get FederationResolver with Limit and Page: %v - alerts: %+v", err, fedResolverWithPage.Alerts)
+	}
+	if !reflect.DeepEqual(fedResolver[1:2], fedResolverWithPage.Response) {
+		t.Error("expected GET FederationResolver with limit = 1, page = 2 to return second result")
+	}
+
+	opts.QueryParameters = url.Values{}
+	opts.QueryParameters.Set("limit", "-2")
+	resp, reqInf, err := TOSession.GetFederationResolvers(opts)
+	if err == nil {
+		t.Error("expected GET FederationResolver to return an error when limit is not bigger than -1")
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+
+	opts.QueryParameters.Set("limit", "1")
+	opts.QueryParameters.Set("offset", "0")
+	resp, reqInf, err = TOSession.GetFederationResolvers(opts)
+	if err == nil {
+		t.Error("expected GET FederationResolver to return an error when offset is not a positive integer")
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+
+	opts.QueryParameters = url.Values{}
+	opts.QueryParameters.Set("limit", "1")
+	opts.QueryParameters.Set("page", "0")
+	resp, reqInf, err = TOSession.GetFederationResolvers(opts)
+	if err == nil {
+		t.Error("expected GET FederationResolver to return an error when page is not a positive integer")
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Fatalf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func SortTestFederationResolver(t *testing.T) {
+	var sortedList []uint
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("orderby", "id")
+	resp, _, err := TOSession.GetFederationResolvers(opts)
+	if err != nil {
+		t.Fatalf("Expected no error, but got %v - alerts: %+v", err, resp.Alerts)
+	}
+	for _, fedRes := range resp.Response {
+		sortedList = append(sortedList, *fedRes.ID)
```

Comment:
This will segfault if TO improperly replies with a `null` or `undefined` `id` property for any Federation Resolver.

Reply:
added null checks.